src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh

```diff
@@ -23,7 +23,7 @@ if [[ "$VSCODE_INJECTION" == "1" ]]; then
 		ZDOTDIR=$VSCODE_ZDOTDIR
 	fi
 
-	if [[ -f $USER_ZDOTDIR/.zsh_history && -z $HISTFILE ]]; then
+	if [[ -f $USER_ZDOTDIR/.zsh_history && -f $HISTFILE ]]; then
 		HISTFILE=$USER_ZDOTDIR/.zsh_history
 	fi
```

Comment:
I forgot why we have this at all? If $HISTFILE is not set, doesn't it use the default? This is the only place we reference HISTFILE

Reply:
with the ZDOTDIR set to $VSCODE_ZDOTDIR above, VSCODE_ZDOTDIR/.zsh_history would run without this

Reply:
I'm not sure why we set the ZDOTDIR back to VSCODE_ZDOTDIR on line 23. Do you recall?

Reply:
since there's no `.zlogout` file, we should be safe to just set $ZDOTDIR=$USER_ZDOTDIR earlier in this file, which would fix this

Reply:
Don't we then need to unset it at the end of the scripts (which depends on if it was login) if it was unset previously/

Reply:
yeah that's what we currently do.